src/list.c

```diff
@@ -40,11 +40,17 @@ void* statsrelay_list_expand(list_t list) {
 }
 
 void statsrelay_list_destroy(list_t list) {
+        if (list == NULL) {
+		return;
+        }
 	free(list->data);
 	free(list);
 }
 
 void statsrelay_list_destroy_full(list_t list) {
+	if (list == NULL) {
```

Comment:
Not sure if we should really check here... At least we won't segfault here but somewhere up, easier to catch bugs